zebra-chain/src/parameters/tests.rs

```diff
@@ -233,3 +233,20 @@ fn branch_id_consistent(network: Network) {
         }
     }
 }
+
+// TODO: split this file in unit.rs and prop.rs
+use hex::{FromHex, ToHex};
+use proptest::prelude::*;
+
+proptest! {
+    #[test]
+    fn branch_id_hex_roundtrip(nu in any::<NetworkUpgrade>()) {
+        zebra_test::init();
+
+        if let Some(branch) = &nu.branch_id() {
+            let hex_branch: String = branch.encode_hex();
+            let new_branch = ConsensusBranchId::from_hex(hex_branch.clone()).expect("hex branch_id should parse");
+            prop_assert_eq!(hex_branch, new_branch.to_string());
```

Comment:
Minor: I'd also add another check, just to be sure.

```suggestion
            prop_assert_eq!(branch, new_branch);
            prop_assert_eq!(hex_branch, new_branch.to_string());
```

To apply this you'll also need to remove the `&` from `&nu.branch_id()`.

Reply:
https://github.com/ZcashFoundation/zebra/pull/3891/commits/c601c8ab3ed0c4b84a64b26cc2aac6507094c962